src/lib/tls/tls_record.cpp

```diff
@@ -340,10 +340,13 @@ Record_Header read_tls_record(secure_vector<uint8_t>& readbuf,
       }
 
    /*
-   This is a little hacky but given how TLS 1.3 versioning works it
-   is probably safe
+   Verify that the record type and record version are within some expected
+   range, so we can quickly reject totally invalid packets.
+
+   The version check is a little hacky but given how TLS 1.3 versioning works
+   this is probably safe
    */
-   if(readbuf[1] != 3)
+   if(readbuf[0] < 20 || readbuf[0] > 23 || readbuf[1] != 3 || readbuf[2] >= 4)
```

Comment:
Content type check will be different in TLS 1.3, where there are additional allowed values: invalid(0) and heartbeat(24). But it will anyway need separate handling/implementation probably.